hw4.Rmd

```diff
@@ -30,43 +37,103 @@ Before doing anything else, just play with the data to get a sense of it.
 (@) The main outcome of interest is *decision* which records whether the judge granted an appeal or not.  In what proprotion of cases did the judge grant an appeal.
 
 ```{r}
-# [your code here]
+# languages less likely to have an appeal granted?
+dat %>%
+  select(language, decision) %>%
+  group_by(language) %>%
+  summarise(yes = mean(decision=="yes")) %>%
+  ggplot(aes(x=language, y=yes)) +
+    geom_bar(stat="identity") +
+    ylab("percent appealed granted")
+
+# cases vs. percent appeal granted per nation:
+dat %>%
+  select(nation, decision) %>%
+  ggplot(aes(x=nation, fill=decision)) +
+    geom_bar() +
+    theme(axis.text.x=element_text(angle=90, hjust=1))
+
 ```
 
 (@) There are 12 different judges in the data. A key question is whether different judges have different rates of granting an appeal.  Make a plot showing the number of cases per judge.
 
 ```{r}
-# [your code here]
+# are the judges fair 50/50 in granting appeal?
+dat %>%
+  select(judge, decision) %>%
+  group_by(judge) %>%
+  summarise(cases = sum(decision)) %>%
+  ggplot(aes(x=judge, y=cases, label=cases)) +
+    geom_bar(stat="identity") +
+    geom_text(vjust=-1) +
+    expand_limits(y=c(0, 100)) +
+    ylab("number of cases")
 ```
 
 (@) Now plot the rate at which the judges grant appeals. 
 
 ```{r}
-# [your code here]
+# are the judges fair 50/50 in granting appeal?
+dat %>%
+  select(judge, decision) %>%
+  group_by(judge) %>%
+  summarise(yes = mean(decision=="yes")) %>%
+  ggplot(aes(x=judge, y=yes)) +
+    geom_bar(stat="identity") +
+    expand_limits(y=c(0, 1)) +
+    ylab("percent of appeals granted")
 ```
 
 (@) Now let's try this with logistic regression because we are going to move to more complicated models.  Fit a logistic regression predicting whetheran appeal will be granted using judge as a categorical predictor. Use Iacobucci as the reference judge to facilitate the comparisons that follow. For more on how to control the reference level of a factor, check out this post on StackOverflow: http://stackoverflow.com/questions/3872070/how-to-force-r-to-use-a-specified-factor-level-as-reference-in-a-regression
 
 ```{r}
-# [your code here]
+# set judge Iacobucci as reference category :
+dat$judge <- relevel(dat$judge, "Iacobucci")
+
+glm(decision ~ judge, data=dat, family="binomial") %>%
+  summary()
 ```
 
 (@) Using the results of the model calculate the predicted probability that Judge Marceau will grant an appeal.
 
 ```{r}
-# [your code here]
+print("When compared to judge Iacobucci, the probability judge Marceau will grant an appeal is:")
+
+glm(decision ~ judge, data=dat, family="binomial") %>%
+  tidy() %>%
+  filter(term == "(Intercept)" | term == "judgeMarceau") %>%
+  select(term, estimate) %>%
+  summarise(percent = round((estimate[2]+estimate[1])*100, digits=2))
```

Comment:
This is not the correct equation for predicting probabilities from logit models.